src/test/steps/signUpSteps.ts

```diff
@@ -0,0 +1,30 @@
+import { When, Then } from '@cucumber/cucumber';
+import { expect } from '@playwright/test';
+import basePage from '../../pages/basePage';
+
+When('User navigates to sign up page by clicking the sign-up button', async function () {
+  await basePage.signUpPage.navigateToRegisterPage();
+});
+
+When('User fill out the account registration form with valid information', async function (dataTable) {
+  const data = dataTable.rowsHash();
+  const country = data['Country'];
+  const currency = data['Default Currency'];
+  const industry = data['Industry'];
+
+
+  await basePage.signUpPage.fillInputs(data['Full Name'], data['Company Name'], data['Phone'],data['CompanyDetails'], data['AccountDetails'])
+  await basePage.signUpPage.genereteEmail();
+  await basePage.signUpPage.agreePrivacyPolicy();
+  await basePage.signUpPage.selectCountry(country);
+  await basePage.signUpPage.selectCurrency(currency);
+  await basePage.signUpPage.selectIndustry(industry);
+  await basePage.signUpPage.clickRegisterButton();
```

Comment:
As I explained yesterday, Step definition should not know about how something is implemented. They should just pass data to lower class, and lower class should perform actions on data.<br>
This step definition will fail the test in case there is no Industry dropdown to be selected from.